File: libCompiler/src/main/java/com/duy/pascal/backend/exceptions/ParsingException.java
package com.duy.pascal.backend.exceptions;


import android.support.annotation.Nullable;

import com.duy.pascal.backend.linenumber.LineInfo;

/**
 * Exception when parse syntax of the program
 */
public class ParsingException extends Exception {

    /**
     * The lineInfo of lineInfo error
     */
    public LineInfo lineInfo;

    public ParsingException(@Nullable LineInfo lineInfo, String message) {
        super(message);
        this.lineInfo = lineInfo;
    }

    public ParsingException(LineInfo lineInfo) {
        super();
        this.lineInfo = lineInfo;
    }

    public LineInfo getLineInfo() {
        return lineInfo;
    }

    public void setLineInfo(LineInfo lineInfo) {
        this.lineInfo = lineInfo;
    }

    @Override
    public String toString() {
        return lineInfo + ":" + getMessage();
    }

    public boolean isAutoFix() {
        return false;
    }
}

File: libCompiler/src/main/java/com/duy/pascal/backend/exceptions/ParsingException.kt
package com.duy.pascal.backend.exceptions


import com.duy.pascal.backend.linenumber.LineInfo

/**
 * Exception when parse syntax of the program
 */
open class ParsingException : Exception {

    /**
     * The lineInfo of lineInfo error
     */
    var lineInfo: LineInfo? = null

    constructor(lineInfo: LineInfo?, message: String) : super(message) {
        this.lineInfo = lineInfo
    }

    constructor(lineInfo: LineInfo?) : super() {
        this.lineInfo = lineInfo
    }

    override fun toString(): String {
        return lineInfo.toString() + ":" + message
    }

    open val isAutoFix: Boolean
        get() = false
}
